networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1202,13 +1200,16 @@ def bellman_ford_predecessor_and_distance(G, source, target=None,
     weight = _weight_function(G, weight)
 
     dist = _bellman_ford(G, [source], weight, pred=pred, dist=dist,
-                         cutoff=cutoff, target=target)
+                         target=target)
     return (pred, dist)
 
 
 def _bellman_ford(G, source, weight, pred=None, paths=None, dist=None,
-                  cutoff=None, target=None):
-    """Relaxation loop for Bellman–Ford algorithm
+                  target=None):
+    """Relaxation loop for Bellman–Ford algorithm.
+
+    This is an implementation of the SPFA variant.
+    See https://en.wikipedia.org/wiki/Shortest_Path_Faster_Algorithm
```

Comment:
I'm not sure if this is the best way to cite the algorithm though.

Reply:
I agree, though there seems to be ambiguity over how to cite it.  It is claimed [on a WCIPEG webpage](https://wcipeg.com/wiki/Shortest_Path_Faster_Algorithm) that Yen's improvements on Bellman-Ford included this algorithm. But wikipedia cites a paper by Duan Fanding. All this can be found from links at the bottom of the wikipedia page, so this citation should be good for our purposes.